### ISSUE TITLE
2019-6-29

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#41]
931.2 - Building blocks of the model
932.2.1 - Clipping the gradients in the optimization loop
933.clip function
934.2.2 - Sampling
935.sample
936.3 - Building the language model
937.3.1 - Gradient descent
938.optimize
939.3.2 - Training the model
